server: TestCheckRestartSafe_Integration failed

server.TestCheckRestartSafe_Integration [failed](https://mesolite.cluster.engflow.com/invocations/default/a7296a02-1d9e-42ae-bceb-ff3bca992732?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyOnNlcnZlcl90ZXN0) on master @ [aa0ff2be33740efaea473d750e286848e988caca](https://github.com/cockroachdb/cockroach/commits/aa0ff2be33740efaea473d750e286848e988caca):


```
=== RUN   TestCheckRestartSafe_Integration
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestCheckRestartSafe_Integration1170966042
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    test_server_shim.go:594: DRPC is enabled (override by TestingGlobalDRPCOption)
    api_v2_test.go:620: 
        	Error Trace:	pkg/server/api_v2_test.go:620
        	            				pkg/server/api_v2_test.go:594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = result is ambiguous: error=DRPC: failed to connect to n2 at 127.0.0.1:36769: initial connection heartbeat failed: dial tcp 127.0.0.1:36769: connect: connection refused [exhausted] (last error: failed to send RPC: sending to all replicas failed; last error: [NotLeaseHolderError] lease held by different store; r42: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):2 seq=2 start=1759953724.834614394,0 exp=1759953730.834198393,0 pro=1759953724.834198393,0 acq=Transfer)
        	Test:       	TestCheckRestartSafe_Integration
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestCheckRestartSafe_Integration1170966042
--- FAIL: TestCheckRestartSafe_Integration (14.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_Integration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55275

Epic CRDB-55382

server.TestCheckRestartSafe_Integration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20604103?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20604103?buildTab=artifacts#/) on master @ [3a6b122a3edeac7622c0b902e4fd50556a212a17](https://github.com/cockroachdb/cockroach/commits/3a6b122a3edeac7622c0b902e4fd50556a212a17):


```
=== RUN   TestCheckRestartSafe_Integration
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/1930f76408fbf7cf03ad9fcbea803b88/logTestCheckRestartSafe_Integration1946570067
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    test_server_shim.go:594: DRPC is enabled (override by TestingGlobalDRPCOption)
    api_v2_test.go:620: 
        	Error Trace:	pkg/server/api_v2_test.go:620
        	            				pkg/server/api_v2_test.go:594
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = result is ambiguous: error=DRPC: failed to connect to n2 at 127.0.0.1:37537: initial connection heartbeat failed: dial tcp 127.0.0.1:37537: connect: connection refused [exhausted] (last error: failed to send RPC: sending to all replicas failed; last error: [NotLeaseHolderError] lease held by different store; r42: replica (n1,s1):1 not lease holder; current lease is repl=(n2,s2):2 seq=2 start=1760559640.041938480,0 type=LeaseExpiration exp=1760559646.041885504,0 pro=1760559640.041885504,0 acq=Transfer)
        	Test:       	TestCheckRestartSafe_Integration
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1930f76408fbf7cf03ad9fcbea803b88/logTestCheckRestartSafe_Integration1946570067
--- FAIL: TestCheckRestartSafe_Integration (15.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCheckRestartSafe_Integration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/api_v2_test.go
+++ b/pkg/server/api_v2_test.go
@@ -31,6 +31,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/metric"
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/errors"
 	"github.com/stretchr/testify/require"
 	yaml "gopkg.in/yaml.v2"
 )
@@ -385,10 +386,16 @@ func TestCheckRestartSafe_Criticality(t *testing.T) {
 	err = drain(ctx, ts1, t)
 	require.NoError(t, err)
 
-	res, err = checkRestartSafe(ctx, ts1.NodeID(), ts1.NodeLiveness().(livenesspb.NodeVitalityInterface), ts1.GetStores().(storeVisitor), 3, false)
-	// Now that we've drained, we're ok to restart
-	require.NoError(t, err)
-	require.True(t, res.IsRestartSafe)
+	testutils.SucceedsSoon(t, func() error {
+		res, err = checkRestartSafe(ctx, ts1.NodeID(), ts1.NodeLiveness().(livenesspb.NodeVitalityInterface), ts1.GetStores().(storeVisitor), 3, false)
+		if err != nil {
+			return err
+		}
+		if !res.IsRestartSafe {
+			return errors.New("expected IsRestartSafe to be true")
+		}
+		return nil
+	})
 }
 
 // TestCheckRestartSafe_RangeStatus verifies that checkRestartSafe is
@@ -575,12 +582,7 @@ func TestCheckRestartSafe_Integration(t *testing.T) {
 	ctx := context.Background()
 	var err error
 
-	testCluster := serverutils.StartCluster(t, 3, base.TestClusterArgs{
-		ServerArgs: base.TestServerArgs{
-			// TODO(155111): Test is flakey with DRPC enabled.
-			DefaultDRPCOption: base.TestDRPCDisabled,
-		},
-	})
+	testCluster := serverutils.StartCluster(t, 3, base.TestClusterArgs{})
 	defer testCluster.Stopper().Stop(ctx)
 
 	ts0 := testCluster.Server(0)

